sormas-backend/src/main/java/de/symeda/sormas/backend/common/messaging/ManualMessageLogService.java

```diff
@@ -0,0 +1,58 @@
+package de.symeda.sormas.backend.common.messaging;
+
+import java.util.List;
+
+import javax.ejb.LocalBean;
+import javax.ejb.Stateless;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.From;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import javax.persistence.criteria.Subquery;
+import javax.validation.constraints.NotNull;
+
+import de.symeda.sormas.api.messaging.MessageType;
+import de.symeda.sormas.backend.caze.Case;
+import de.symeda.sormas.backend.common.AbstractAdoService;
+import de.symeda.sormas.backend.person.Person;
+
+@Stateless
+@LocalBean
+public class ManualMessageLogService extends AbstractAdoService<ManualMessageLog> {
+
+	public static final int MANUAL_MESSAGE_LOG_LIMIT = 5;
+
+	public ManualMessageLogService() {
+		super(ManualMessageLog.class);
+	}
+
+	@Override
+	public Predicate createUserFilter(CriteriaBuilder cb, CriteriaQuery cq, From<?, ManualMessageLog> from) {
+		return null;
+	}
+
+	public List<ManualMessageLog> getByCaseUuid(@NotNull String caseUuid, MessageType messageType) {
+
+		final CriteriaBuilder cb = em.getCriteriaBuilder();
+		final CriteriaQuery<ManualMessageLog> cq = cb.createQuery(ManualMessageLog.class);
+		final Root<ManualMessageLog> manualMessageLogRoot = cq.from(ManualMessageLog.class);
+
+		final Subquery<Person> casePersonSubQuery = cq.subquery(Person.class);
```

Comment:
This seems very unperformant to me, especially because you're selecting the whole person which loads a lot of completely unnecessary data from the database. Why not just use joins to compare the ID of the case person to the ID of the message log person?

On another note, since you've added the message log as a lazy property to the person anyway, you could just use personJoin.get(Person.MANUAL_MESSAGE_LOGS) if I'm not mistaken.

Reply:
rewritten it to select only the id in the subquery